stdlib/public/core/StringLegacy.swift

```diff
@@ -33,14 +33,25 @@ extension String {
       self = count == 0 ? "" : repeatedValue
       return
     }
+    guard !repeatedValue.isEmpty else {
+      self = ""
+      return
+    }
 
-    // TODO(String performance): We can directly call appendInPlace
-    var result = String()
-    result.reserveCapacity(repeatedValue._guts.count &* count)
-    for _ in 0..<count {
-      result += repeatedValue
+    var repeatedValue = repeatedValue
+    self = repeatedValue.withUTF8 { repeatedUTF8 in
+      String(_uninitializedCapacity: repeatedUTF8.count * count) { buffer in
+        var total = 0
+        for i in 0..<count {
+          let offset = i &* repeatedUTF8.count
+          let range = offset ..< offset + repeatedUTF8.count
+          _ = UnsafeMutableBufferPointer(rebasing: buffer[range])
+            .initialize(from: repeatedUTF8)
+          total += range.count
+        }
+        return total
```

Comment:
If you're going to track total separately from the uninitialized capacity, could you add an assert that they're the same?

Reply:
done